Tools/scripts/decode-ICSR.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+'''
+decode an stm32 ICSR register value
+'''
+
+import sys
+import optparse
+
+def num(s):
+    try:
+        return int(s)
+    except ValueError:
+        return int(s, 16)
+
+
+parser = optparse.OptionParser(__file__)
+
+opts, args = parser.parse_args()
+
+if len(args) == 0:
+    print(parser.usage)
+    sys.exit(0)
+
+ICSR = num(args[0])
+
+# https://www.st.com/content/ccc/resource/technical/document/programming_manual/6c/3a/cb/e7/e4/ea/44/9b/DM00046982.pdf/files/DM00046982.pdf/jcr:content/translations/en.DM00046982.pdf
+# page 255
+
+def decoder_m4_vectactive(value):
+    exceptions = {
+        0: "Thread mode",
+        1: "Reserved",
+        2: "NMI",
+        3: "Hard fault",
+        4: "Memory management fault",
+        5: "Bus fault",
+        6: "Usage fault",
+        7: "Reserved....",
+        10: "Reserved",
+        11: "SVCall",
+        12: "Reserved for Debug",
+        13: "Reserved",
+        14: "PendSV",
+        15: "SysTick",
+    }
+    if value in exceptions:
+        exception = "%s" % str(exceptions[value])
+    else:
+        exception = "IRQ%u" % (value - 16)
+
+    print(" (%s)" % exception),
```

Comment:
trailling comma here

Reply:
In Python2 the trailing comma mean, "don't add a newline".

That appears to have gone away in Python3, as you noted elsewhere.